### PR TITLE
Updates to the web interface for final announcement to junta

### PR DESCRIPTION
**Pull request by yashsriv**
Features added:
* Improved speed of animations by not using masonry anymore
* Add support polyfills for IE 9+ so that it open on Windows Phones / old Safari as well
* Add a box displaying number of search results
* Improve help text
* Converted the application to a PWA!!

Please review commit by commit for a proper idea of the flow.

This is currently deployed at https://search.pclub.in if you want to see the final product.

### CODE REVIEW

File: frontend/src/app/components/help-dialog/help-dialog.component.html [hunk 1]
```diff
@@ -1,8 +1,18 @@
...
+  You can customise the image shown here by placing a custom image in your iitk webhome folder called
   <code>dp.jpg</code>/<code>dp.png</code> such that going to <code>http://home.iitk.ac.in/~yourusername/dp</code>
   opens up that particular picture.
+  </p>
+
+  <h3> How do I update the data shown here? </h3>
+  <p>
+    The data here is scraped from the <a href="http://oa.cc.iitk.ac.in/Oa/Jsp/Main_Frameset.jsp">Office Automation Portal</a>.
+    The data there can be updated via the <code> Login Based Services > Student Profile > PI form </code>. If you have had
+    a branch change, please go to the ID Cell and update your ID Card to update the your branch.
```
Comment (milindl) [inline frontend/src/app/components/help-dialog/help-dialog.component.html]: Extra 'the' after 'update'

File: frontend/src/app/components/search/search.component.ts [hunk 2]
```diff
@@ -139,12 +142,48 @@ export class SearchComponent implements OnInit {
     this.dialog.open(MailDialogComponent, {
       data: {
         mails: this.allResults.map((val) => val.u + '@iitk.ac.in')
-      }
+      },
+      panelClass: 'centered-dialog',
```
Comment (milindl) [inline frontend/src/app/components/search/search.component.ts]: Looks like a trailing comma?

File: frontend/src/app/components/search/search.component.ts [hunk 3]
```diff
@@ -139,12 +142,48 @@ export class SearchComponent implements OnInit {
...
     });
   }
 
   showHelpDialog() {
-    this.dialog.open(HelpDialogComponent);
+    this.dialog.open(HelpDialogComponent, {
+      panelClass: 'centered-dialog',
```
Comment (milindl) [inline frontend/src/app/components/search/search.component.ts]: Trailing comma?

File: frontend/src/manifest.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      "src": "/assets/images/icons/android-chrome-512x512.png",
+      "sizes": "512x512",
+      "type": "image/png"
+    }
+  ],
+  "theme_color": "#3f51b5",
+  "background_color": "#ffffff",
+  "display": "standalone",
+  "orientation": "portrait",
+  "Scope": "/",
```
Comment (milindl) [inline frontend/src/manifest.json]: Nit: Scope -> scope

File: frontend/src/polyfills.ts [hunk 5]
```diff
@@ -19,25 +19,24 @@
  */
 
 /** IE9, IE10 and IE11 requires all of the following polyfills. **/
```
Comment (milindl) [inline frontend/src/polyfills.ts]: Nit: require

Comment (milindl) [inline frontend/src/polyfills.ts]: This seems to be an issue with the entire file.

Comment (yashsriv) [inline frontend/src/polyfills.ts]: This file was auto-generated by ng cli... Please file an issue there 😛

**Comment by yashsriv** [comment]
Updating with required changes squashed into the respective commits where the errors were introduced.

**Pull request merged**